react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js

```diff
@@ -8,16 +8,31 @@ const initialState = {
 
 function reducer(state = initialState, action) {
   switch (action.type) {
-    case actions.GET_BOOKS: // TODO to implement the logic
-      return { ...state };
-    case actions.ADD_TO_CART: // TODO to implement the logic
-      return { ...state };
-    case actions.ADD_ITEM: // TODO to implement the logic
-      return { ...state };
-    case actions.REMOVE_ITEM: // TODO to implement the logic
-      return { ...state };
-    case actions.SEARCH_ITEM: // TODO to implement the logic
-      return { ...state };
+    case actions.GET_BOOKS:
+      return { ...state, books: action.payload, originalData: action.payload };
+    case actions.ADD_TO_CART:
+      return { ...state, bookSelected: [...state.bookSelected, action.payload] };
+    case actions.ADD_ITEM: {
+      const addItem = state.bookSelected.find(book => book.id === action.payload);
+      addItem.quantity += 1;
+      return { ...state, bookSelected: [...state.bookSelected] };
```

Comment:
i dont get this, you are mutating the state? because `state.bookSelected.find()` is not the best way to achieve this, its better to make a copy of book selected, sum the quantity and return the new boook selected